src/x86_64.rs

```diff
@@ -0,0 +1,331 @@
+// Copyright 2018-2019 CrowdStrike, Inc.
```

Comment:
Design nit: I would split the code here in two files:
1. One file - windows specific structures.
2. lib.rs/other: The logic of selecting between the linux vs windows structures.

For example we could have a windows module. In `windows/mod.rs` we would have the definition of windows specific structures and their tests. This would end up being the `kvm-bindings` of windows. The nice thing with this approach is that you don't need to label each individual structure/test with the conditional compilation macros.

```rust
#[cfg!(windows)

pub struct LapicState {
    pub regs: [::std::os::raw::c_char; 4096usize],
}

impl Default for LapicState {
    fn default() -> Self {
        unsafe { ::std::mem::zeroed() }
    }
}

impl ::std::fmt::Debug for LapicState {
    fn fmt(&self, fmt: &mut ::std::fmt::Formatter) -> ::std::fmt::Result {
        self.regs[..].fmt(fmt)
    }
}

....

mod tests {

#[test]
fn vcpu_test_layout_lapic_state() {
    assert_eq!(
        ::std::mem::size_of::<LapicState>(),
        1024usize,
        concat!("Size of: ", stringify!(LapicState))
    );
    assert_eq!(
        ::std::mem::align_of::<LapicState>(),
        1usize,
        concat!("Alignment of ", stringify!(LapicState))
    );
    assert_eq!(
        unsafe { &(*(::std::ptr::null::<LapicState>())).regs as *const _ as usize },
        0usize,
        concat!(
            "Offset of field: ",
            stringify!(LapicState),
            "::",
            stringify!(regs)
        )
    );
}
}
``` 

Then the responsibility for choosing between the appropriate structure depending on the operating system would move to `src/lib.rs`:
```
#[cfg(windows)]
mod windows;

#[cfg(windows)]
pub use windows::LapicState;
#[cfg(unix)]
pub use kvm_bindings::kvm_lapic_state as LapicState;

// kvm_msrs looks the same on both platforms; just export it as Msrs
pub use kvm_bindings::kvm_msrs as Msrs;
....
```

Reply:
It looks like you're operating off an older commit, before I redefined all the structures for Windows compilation. But your point not only still stands, it's probably even more important now that there is a ton of conditional compilation statements with mixed Windows/Unix definitions. I'm working on splitting it out as you've suggested.

Reply:
Split Windows definitions into a separate file as suggested